### ISSUE TITLE
"Loading..." on smartphone

### ISSUE THREAD

**Issue by romu70** [Laverna/laverna#349]
Hi,
Laverna works perfectly on my 2 PCs, but doesn't work anymore on my smartphone (Gecko based browser), the page displays "Loading..." and I never get the app.

Any idea?

**Comment by wwebfor** [comment]
Strange. I just checked on Android version of Firefox, and it worked. Do you use Firefox?

**Comment by romu70** [comment]
Do, I run Sailfish OS which runs a gecko based browser, not Firefox in itself. I've tried to clear cache and cookies, I restarted the browser, no luck, still "loading...".

**Comment by claell** [comment]
Is it possible to delete the browser data like on Android in the settings?

**Comment by romu70** [comment]
Yes, I can choose which category of data have to be cleared. I chose cookies and cache, but some other can be chosen.

**Comment by claell** [comment]
I meant in the system settings. On Android you could delete every personal data of your browser so it acts as if it was just installed. That might help, but I don't know whether there is such option on Sailfish and whether you want to delete everything.

**Comment by romu70** [comment]
I'll search, but deleting everything is indeed not what I want. What's strange is it still worked like a charm 2 or 3 weeks ago.

**Comment by romu70** [comment]
My Laverna is setup to sync with my DropBox account. Few hours ago, I changed the email address attached to DropBox, could this be the cause of the issue?

Anyway, I deleted everything on my phone, no change. But, if I browse in private mode, it works. Do you know how to fix this? Even if I've to setup again the Dropbox association.

**Comment by claell** [comment]
It looks a bit like issue #300.

**Comment by wwwredfish** [comment]
I will try to install Sailfish OS later on my tablet and test.

**Comment by wwwredfish** [comment]
I am getting the same loading screen.  We will try to fix it as soon as we figure out what is causing it. Thanks for reporting this bug.

**Comment by romu70** [comment]
Great, a bug reproduced is bug on the way to be fixed!

**Comment by bencolombini** [comment]
I also have this problem of the "Loading..." that never goes anywhere.  For me it occurs on Pale Moon (a fork of Firefox) on Linux (xubuntu desktop linux).

Using https://laverna.cc/dist/ WORKED for me.  I'll use this to work-around for now.

It also works for me in Firefox and Chromium.

It also works in Private Browsing mode in Pale Moon (big possible clue)

I can use the /dist/ version for now, but would still like to help get the bug resolved so I can use Pale Moon on the main version.  These attempts did not fix it for me:

- Trying https://laverna.cc/nightly/
- Clearing offline storage, which it did manage to write to
- Checking permissions on the IndexedDB offline storage folder (I owned them and it was able to write and did make a laverna.cc related folder
- Running in the browser console
  `var req = indexedDB.open('notes-db'); req.onsuccess = function() {console.log('success');}; req.onerror = function(e) {console.error('Error', e);}`
  returned "success" but did not fix it
- there were no cookies with the word "laverna" in them to clear
- clearing cache didn't help.  I refuse to clear all cookies and history at this time, sorry

I hope this helps to resolve the issue or at least let others know how to work around.  Thank you, and I look forward to getting off Evernote thanks to your efforts!

**Comment by romu70** [comment]
https://laverna.cc/dist/ also works for me but shows notes which are pretty outdated, side question, where do those notes come from?

https://laverna.cc/nightly/ has the "Loading..." issue too.

What strange is in private mode, I can go to https://laverna.cc/ and click "Start using now" and the application starts indeed. I also suspect something with the storage and settings.

**Comment by wwwredfish** [comment]
Thanks for the information! It seems like it is happening because of webworkers. We will try to fix it asap. Hopefully Pale Moon and Sailfish's browser share the same problem.

P.S. Pale Moon's debugging tools helped a lot to figure out what is happening.

**Comment by SwagMuffinMcYoloPants** [comment]
I have the same issue but it's on my iOS device.  https://laverna.cc/dist/ works on my iPhone.
